#!/usr/bin/env python
PACKAGE = "visual_contact"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("GF_pyr_scale",    double_t,    0, "pyramid scale", 0.5,  0, 1)
gen.add("GF_levels", int_t, 0, "pyramid layers",    3, 1,   10)
gen.add("GF_winsize",    int_t,    0, "averaging window size",  15, 3, 25)
gen.add("GF_iterations",   int_t,   0, "number of iterations per level",  3, 1, 10)
gen.add("GF_poly_n",   int_t,   0, "size of pixel neighborhood",  5, 3, 15)
gen.add("GF_poly_s",   double_t,   0, "Gaussian smoothig standard deviation",  1.2, 1, 5)
gen.add("GF_disp_scale",   int_t,   0, "optical flow visualization scale factor",  10, 1, 25)
gen.add("GF_disp_step",   int_t,   0, "optical flow visualization step factor",  16, 1, 100)
gen.add("TTC_disp_scale",   int_t,   0, "TTC map visualization scale factor",  255, 100, 1000)

exit(gen.generate(PACKAGE, "visual_contact", "visual_contact"))
